### ISSUE TITLE
Lock wheel

### ISSUE THREAD

**Issue by Erudition** [Wheellog/Wheellog.Android#490]
**Is your feature request related to a problem? Please describe.**
No ability to lock/unlock wheel

**Describe the solution you'd like**
A lock icon in the top right corner that I can 1-tap unlock/lock the wheel

**Describe alternatives you've considered**
I use EUC world for this feature, but they bury it too much, so not as often as I should.

**Comment by palachzzz** [comment]
Which EUC do you have?

**Comment by n-osennij** [comment]
+1 
kugoo u2 (kingsong 16s)

**Comment by palachzzz** [comment]
Wheel lock is not supported for KS. And don't see real reason to support.

**Comment by Erudition** [comment]
I have a different model, the [King Song S16 Pro](https://www.ewheels.com/product/king-song-s16-pro-1480wh-battery-3000w-motor-6kw-peak/). Wheel Lock is a feature and supported in other apps.

When you say "I don't see a real reason to support", what do you mean? 

One reason for this, like other missing features, would be to further the goal of WheelLog having comprehensive control over compatible wheels -- such that the proprietary manufacturer apps aren't required.

Are there design goals stated somewhere that this conflicts with? If you can point me to them, I will make sure not to submit any feature requests that are out of scope.

**Comment by palachzzz** [comment]
> When you say "I don't see a real reason to support", what do you mean?
This feature is almost useless from my point of view, and I don't know anyone who is using that, moreover, many beginers confused with it, when they acasionaly turn it on, and was not able to ride anymore.

> Are there design goals stated somewhere that this conflicts with? If you can point me to them, I will make sure not to submit any feature requests that are out of scope.
Nothing like that, Wheellog is opensource software, and made in our spare time. So I decide by myself what is interesting to implement, and what is not. You can implement feature you want by yourself, and then send push request, we will decide to approve it or not. Sometimes we can't approve, as it was made for only one wheel/person/etc, or, even braking some other functionality...
Regarding what I decide to implement, and what not to implement... 
Can I manage my time as I want, what do you think? Or should I rush for every dubious feature that two people in the world need?
When I see any request, I'm trying to answer following questions:
1. Is this feature useful for many people?
2. Is this feature questionable from the point of view of physics, user interface, implementation?
3. Do I want and have time to implement it?

At some point I was really interested to implement control of every wheel, but for now, protocols become too complex, and the number of wheels are increasing each year. It takes to much time to implement every new feature for new wheel, and sometimes they may backport it for old wheels. 

The main feature of Wheellog, which you can get from the name of App - is to log data like blackbox. So, I'm doing everything to keep Wheellog updated to be able to read protocols of every new wheel, to make it possible to work as blackbox for it. All other features is not main scope.

**Comment by Erudition** [comment]
Hi @palachzzz , you of course are free to implement only features that you find interesting. That is different from rejecting feature requests, though. In many communities, a feature request that has not been implemented remains an open issue, rather than being closed because one person does not want to do it. I opened this issue under that premise.

If it's your policy to reject features you don't personally want, that's okay, but it's good to be explicit about it, especially if someone else does the work to implement it (like you said) only to find the pull request is rejected because of your preference for scope. I don't have enough Android development experience to do it myself, but if I did, I probably wouldn't want to go so far as to maintain and publish a fork.

> At some point I was really interested to implement control of every wheel

Thanks for providing the update about the scope, specifically that you only envision Wheellog being useful for data logging and not being a comprehensive tool comparable to EUC World.

**Comment by palachzzz** [comment]
Ok, I agree with you, that if I don't have time/intention to implement it - it doesn't mean that issue should be closed.

> Interesting report. I live in a city, so every EUC rider I know locks up their wheel in some way when parking it somewhere out of sight - either with the software lock when an alarm would be sufficient or with a cable lock (when leaving it further away) or both.

I hope we are still talking about software lock, as we are on github. So, what the point of locking the wheel by software, when everyone with EUC World can unlock it? (or Wheellog in case of feature to be implemented)

**Comment by Erudition** [comment]
Yeah, a software lock. One that arms the device's alarm, so that anyone who touches it is treated to a blaring siren and flashing lights. Perfect for when I need to be separated from my wheel by sight but not by earshot, for a duration so brief that chaining it up would be cumbersome and I'd be otherwise tempted to not employ any security at all.

For example, out in nature and I need to use a portajohn. I can't fit it in with me, and there's nothing nearby to chain it to... but even if there was it wouldn't feel worth the effort given that no one is around. If someone did come swipe it without the alarm armed, they could get away silently - but with the alarm, not a chance. And it's a simple tap on the watch, so why not. 

Nearly everyone in my country doesn't know what an EUC is, let alone how one works, let alone owning one, let alone having an app like EUC World at the ready and the know-how to stealthily pair their phone to my EUC without triggering the alarm. Plus, chances are if they meet all those requirements, they own an EUC themselves... not likely a thief. I'd say we're talking about such a contrived and rare potential thief that it's more likely to have a passerby with the tools to cut open whatever physical locks I could use. Not that they'd be too interested in doing that once the alarm goes off on them...
